src/main/java/com/teamfive/disscard/DissCardController.java

```diff
@@ -14,98 +15,101 @@
+     * @return A list of JSON objects representing cards.
      */
     @GetMapping("/card/list")
     @ResponseBody
     public List<Card> fetchAllCards() {
+        log.info("Fetching all cards.");
         return cardService.getAll();
     }
 
     /**
-     * Fetches a card's info based on its ID
-     * @param id id of a card registered with DissCard
-     * @return A JSON object representing a card
+     * Fetches a card's info based on its ID.
+     * @param id ID of a card registered with DissCard.
+     * @return A ResponseEntity containing a JSON object representing a card.
      */
     @GetMapping("/card/info/{id}")
     public ResponseEntity<Card> fetchCardById(@PathVariable("id") int id) {
+        log.info("Fetching card with ID: {}", id);
         Card fetchedCard = cardService.getById(id);
-        HttpHeaders headers = new HttpHeaders();
-        headers.setContentType(MediaType.APPLICATION_JSON);
-        return new ResponseEntity<>(fetchedCard, headers, HttpStatus.OK);
+        return buildResponse(fetchedCard);
     }
 
     /**
-     * Searches DissCard's systems for a list of cards that match the specified keyword
-     * @param keyword String used to filter cards
-     * @return A list of card JSON objects that match the specified keyword
+     * Searches DissCard's systems for a list of cards that match the specified keyword.
+     * @param keyword String used to filter cards.
+     * @return A ResponseEntity containing a list of card JSON objects that match the specified keyword.
      */
     @GetMapping("/card/search/{keyword}")
     public ResponseEntity<List<Card>> fetchCardsByKeyword(@PathVariable("keyword") String keyword) {
+        log.info("Searching for cards with keyword: {}", keyword);
         List<Card> fetchedCards = cardService.searchByName(keyword);
-        HttpHeaders headers = new HttpHeaders();
-        headers.setContentType(MediaType.APPLICATION_JSON);
-        return new ResponseEntity<>(fetchedCards, headers, HttpStatus.OK);
+        return buildResponse(fetchedCards);
     }
 
     /**
-     * Adds a card to DissCard's systems
-     * @param card Card to be added
-     * @return A JSON object representing the card that was added
+     * Adds a card to DissCard's systems.
+     * @param card Card to be added.
+     * @return A ResponseEntity containing a JSON object representing the card that was added.
      */
     @PostMapping(value="/card/add", consumes="application/json", produces="application/json")
     public ResponseEntity<Card> addCard(@RequestBody Card card) {
-        Card newCard = null;
-        HttpHeaders headers = new HttpHeaders();
+        log.info("Adding new card: {}", card);
         try {
-            newCard = cardService.save(card);
+            Card newCard = cardService.save(card);
+            return buildResponse(newCard);
+        } catch (Exception e) {
+            log.error("Error adding card: {}", e.getMessage());
```

Comment:
HTTP Status Code: The status code for a successful addition of a card was changed from HttpStatus.OK to HttpStatus.CREATED (201). This better represents the outcome of the operation, which is the creation of a new resource.